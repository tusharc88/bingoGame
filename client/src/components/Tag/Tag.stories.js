import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered';
import { withKnobs, number, boolean } from '@storybook/addon-knobs';
import Tag from './index';

storiesOf('Tag', module)
  .addDecorator(withKnobs)
  .addDecorator(centered)
  .addWithInfo('default', <Tag />, () => (
    <Tag
      dimension={number('dimension', 15)}
      hasBorder={boolean('hasBorder', false)}
    >
      {number('children', 1)}
    </Tag>
  ));
