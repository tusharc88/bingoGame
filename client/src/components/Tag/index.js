import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const TagWrapper = styled.div`
  height: ${props => `${props.dimension * 2}px`};
  width: ${props => `${props.dimension * 3}px`};
  background-color: #edf0f3;
  text-align: center;
  vertical-align: middle;
  font-size: ${props => `${props.dimension}px`};
  line-height: ${props => `${props.dimension * 2}px`};
  border-radius: 5px;
  ${props => props.hasBorder && `border: 2px solid #3ecedf;`};
`;

const Tag = ({ children, dimension, hasBorder }) => (
  <TagWrapper dimension={dimension} hasBorder={hasBorder}>
    {children}
  </TagWrapper>
);

Tag.propTypes = {
  children: PropTypes.number.isRequired,
  dimension: PropTypes.number.isRequired,
  hasBorder: PropTypes.bool,
};

Tag.defaultProps = {
  hasBorder: false,
};

export default Tag;
