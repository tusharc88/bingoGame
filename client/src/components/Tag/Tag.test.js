import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Tag from './index';

test('Renders with default props', () => {
  const tree = renderer.create(<Tag dimension={15}>{1}</Tag>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders with hasBorder prop', () => {
  const tree = renderer
    .create(
      <Tag dimension={15} hasBorder>
        {1}
      </Tag>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
