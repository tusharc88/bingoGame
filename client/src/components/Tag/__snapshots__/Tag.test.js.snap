// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with default props 1`] = `
.c0 {
  height: 30px;
  width: 45px;
  background-color: #edf0f3;
  text-align: center;
  vertical-align: middle;
  font-size: 15px;
  line-height: 30px;
  border-radius: 5px;
}

<div
  className="c0"
>
  1
</div>
`;

exports[`Renders with hasBorder prop 1`] = `
.c0 {
  height: 30px;
  width: 45px;
  background-color: #edf0f3;
  text-align: center;
  vertical-align: middle;
  font-size: 15px;
  line-height: 30px;
  border-radius: 5px;
  border: 2px solid #3ecedf;
}

<div
  className="c0"
>
  1
</div>
`;
